nixos/lib/make-ext4-fs.nix

```diff
@@ -82,10 +87,17 @@ pkgs.stdenv.mkDerivation {
         size=$(( blocks - ''${free##*:} + fudge ))
 
         echo "Resizing from $blocks blocks to $size blocks. (~ $((size*blocksize/1024/1024))MiB)"
-        EXT2FS_NO_MTAB_OK=yes resize2fs $out -f $size
+        EXT2FS_NO_MTAB_OK=yes resize2fs ./$img -f $size
       )
 
       # And a final fsck, because of the previous truncating.
-      fsck.ext4 -n -f $out
+      fsck.ext4 -n -f ./$img
+
+      if [ ${builtins.toString compressImage} ]; then
+        echo "Compressing image"
+        zstd -v ./$img -o $out
+      else
+        cp ./$img $out
+      fi
```

Comment:
I'm not sure it warrants a change, but the condition could have been handled nix-side

```
${if compressImage then ''
  echo "Compressing image"
  zstd -v ./$img -o $out
'' else ''
  cp ./$img $out
''}
```